src/streamlink/plugins/cmmedia.py

```diff
@@ -0,0 +1,121 @@
+    partner_ids_re = re.compile(r"/p/(\d+)/sp/(\d+)/")
+
+    @staticmethod
+    def is_restricted(data):
+        items = {k: v for k, v in data.items() if k.startswith("is")}
+        for k, v in items.items():
+            log.debug(f"{k}: {v}")
+
+        restrictions = [k for k, v in items.items() if v]
+        if restrictions:
+            log.error(f"This site is restricted: ({', '.join(restrictions)})")
+            return True
+
+        return False
+
+    def _get_streams(self):
+        parsed_html = self.session.http.get(self.url, schema=validate.Schema(validate.parse_html()))
+
+        iframe_url = validate.validate(validate.Schema(
+            validate.xml_xpath_string(".//script[contains(text(), '/embedIframeJs/')]/text()"),
+            validate.any(None, validate.all(
+                validate.transform(self.iframe_url_re.search),
+                validate.any(None, validate.all(validate.get(1), validate.url())),
+            )),
+        ), parsed_html)
+
+        if not iframe_url:
+            return
+
+        m = self.partner_ids_re.search(iframe_url)
+        if not m:
+            log.error("Failed to find partner IDs in IFRAME URL")
+            return
+
+        p = m.group(1)
+        sp = m.group(2)
+
+        json = self.session.http.get(iframe_url, schema=validate.Schema(
+            validate.transform(self.json_re.search),
+            validate.any(None, validate.all(
+                validate.get(1),
+                validate.transform(self.unescape_quotes),
+                validate.parse_json(),
+                validate.any({
+                    "entryResult": {
+                        "contextData": {
+                            "isSiteRestricted": bool,
+                            "isCountryRestricted": bool,
+                            "isSessionRestricted": bool,
+                            "isIpAddressRestricted": bool,
+                            "isUserAgentRestricted": bool,
+                            "flavorAssets": [{
+                                "id": str,
+                            }],
+                        },
+                        "meta": {
+                            "id": str,
+                            "name": str,
+                            "categories": validate.any(None, str),
+                        },
+                    },
+                }, {"error": str}),
+            )),
+        ))
+
+        if not json:
+            return
+
+        if "error" in json:
+            log.error(f"API error: {json['error']}")
+            return
+
+        json = json.get("entryResult")
+
+        if self.is_restricted(json["contextData"]):
+            return
+
+        self.id = json["meta"]["id"]
+        self.title = json["meta"]["name"]
+        author = validate.validate(validate.Schema(
+            validate.xml_xpath_string(".//h1[@class='btn-title']/text()"),
+        ), parsed_html)
+        if author:
+            self.author = author
```

Comment:
```suggestion
        self.author = validate.validate(validate.Schema(
            validate.xml_xpath_string(".//h1[contains(@class, 'btn-title')]/text()"),
        ), parsed_html)
```

`validate.xml_xpath_string` can only return `None` or `str`, and `self.author` is initialized as `None`, so the return value of the validation schema will always fit.

`@class=` compares the entire attribute value, not just individual class names (in case there are more or could be added in the future). lxml supports CSS selectors where this wouldn't be a problem, but that requires adding another dependency, which is not worth it. XPath is unfortunately a bit limited here and requires a `contains()` call.

Reply:
Ah, I was using exactly that in my very first push!  I changed it, thinking it would be better, but not realising that it applied to the entire class attribute.  That's good to know, thanks.